### ISSUE TITLE
Trying to load a project with Sdk.WindowDesktop fails.

### ISSUE THREAD

**Issue by vpenades** [microsoft/MSBuildLocator#120]
I've been trying to load a csproj using new `Microsoft.Build.Evaluation.Project("project.csproj", null, null");` and it throws this exception:

The specified SDK "Microsoft.NET.Sdk.WindowsDesktop" could not be found.
Source: Microsoft.Build
Error: MSB4236
MSBuild.CouldNotResolveSdk

The loaded csproj looks like this:

```xml
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">

  <PropertyGroup>
    <TargetFramework>net471</TargetFramework>
    <UseWPF>true</UseWPF>
  </PropertyGroup>

</Project>
```

_Notice it's using net471 framework, not netcoreapp3.1._

Prior loading the file, we register the default locator: `Microsoft.Build.Locator.MSBuildLocator.RegisterDefaults();`

Which reports it's using MSBuild DotNetSDK at C:\Program Files\dotnet\sdk\2.2.300\

Other projects using Microsoft.NET.Sdk just load fine, so it seems it's only affecting Microsoft.NET.Sdk.WindowsDesktop.

VisualStudio2019 is up to date to the latest release version.

The application hosting the Locator library is a netcoreapp3.1 itself.

**Comment by ghelyar** [comment]
This is not a particularly nice workaround because it makes the library pretty pointless, as you have to tell the "locator" library exactly where to look, but you can do this from a .NET Core app, and it will be able to load both .NET Core and .NET Framework projects:
```csharp
MSBuildLocator.RegisterMSBuildPath(@"C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin");
```

Seems that `DiscoveryType.DeveloperConsole` and `DiscoveryType.VisualStudioSetup` only work on the .NET Framework version of this package, and `DiscoveryType.DotNetSdk` only works on the .NET Core version of this package, so this returns completely different results depending on the target framework of the project that is referencing the package:
```csharp
foreach (var instance in MSBuildLocator.QueryVisualStudioInstances(
    new VisualStudioInstanceQueryOptions
    {
        DiscoveryTypes = DiscoveryType.DeveloperConsole | DiscoveryType.DotNetSdk | DiscoveryType.VisualStudioSetup
    }))
{
    Console.WriteLine($"{instance.DiscoveryType} {instance.MSBuildPath}");
}
```

But if you take the paths from either of them and just pass them to `MSBuildLocator.RegisterMSBuildPath`, they work in the other, so it's a bit annoying that it doesn't just find all the same paths regardless of target framework. You can specify `DiscoveryType` but it's mostly pointless, because it ignores it when finding them, and then only uses it to throw away results afterwards.

**Comment by vpenades** [comment]
The problem with that approach is that our project runs on different environments, and it's not guaranteed the path to pass to RegisterMSBuildPath is the same on all machines (specially those with different versions of VS)

**Comment by ghelyar** [comment]
Yes, it's not ideal, and I'm not making any excuses for this library. I'm just a consumer of this library, and I'm sharing my workaround.

Exactly how you get the path to the directory containing msbuild will depend on your environments. For my usage, I can just pass it in as an argument. You might be able to search your path environment variables, or use `where.exe msbuild` to find it, or save it to an environment variable on the machine, or you might be able to do something like this:
```csharp
var msbuildDirectories = new[]
    {
        @"C:\Program Files (x86)\Microsoft Visual Studio",
        @"C:\Program Files\Microsoft Visual Studio"
    }
    .Where(Directory.Exists)
    .SelectMany(path => Directory.EnumerateFiles(path, "msbuild.exe", SearchOption.AllDirectories))
    .Select(Path.GetDirectoryName);
```

Two overloads exist:
* `MSBuildLocator.RegisterMSBuildPath(string)`
* `MSBuildLocator.RegisterMSBuildPath(string[])`

So you could either `.ToArray()` this or `.First()` it or `.Last()` it etc.
